### PR TITLE
Install into ZDOTDIR if set

### LINKED ISSUE #9001: Installer does not respect $ZDOTDIR
**Describe the bug**
`tools/install.sh` installs into `~` even when `$ZDOTDIR` is set.

**To Reproduce**
1. `cd "$(mktemp -d)"`
2. `ZDOTDIR=$PWD zsh -d`
3. `sh -c "$(curl -fsSL https://raw.githubusercontent.com/ohmyzsh/ohmyzsh/master/tools/install.sh)"`

**Expected behavior**
OMZ installs into `$ZDOTDIR`.

**Actual behavior**
OMZ installs into `~`.

**Desktop (please complete the following information):**
 - OS / Distro: macOS
 - Latest ohmyzsh Update?: Yes
 - ZSH Version: 5.8
 - Terminal emulator: Apple Terminal

### LINKED ISSUE #10479: Install script does not account for .zshrc in ZDOTDIR
### Describe the bug

When zsh configuration is using the environment variable ZDOTDIR, the oh my zsh install script incorrectly considers no zsh config file is present.

### Steps to reproduce

1. Create  `~/.zshenv` with content:

```
ZDOTDIR=${HOME}/.config/zsh
```

2. Put actual zsh config at ~/.config/zsh/.zshrc

**Incorrect Behavior:**

3. When installing with `curl`, the install script considers that no .zshrc exists and creates one in $HOME:

```
Looking for an existing zsh config...
Using the Oh My Zsh template file and adding it to `~/.zshrc`.
```

### Expected behavior

Oh my zsh should put its configuration in `$ZDOTDIR/.zshrc` and not create another .zshrc.

### Screenshots and recordings

_No response_

### OS / Linux distribution

Ubuntu 20.04.2. LTS 

### Zsh version

5.8

### Oh My Zsh version

Install script 2021-12-04

### Terminal emulator

Terminator

### If using WSL on Windows, which version of WSL

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by alexpjohnson**
## Standards checklist:

- [X] The PR title is descriptive.
- [X] The PR doesn't replicate another PR which is already open.
- [X] I have read the contribution guide and followed all the instructions.
- [X] The code follows the code style guide detailed in the wiki.
- [X] The code is mine or it's from somewhere with an MIT-compatible license.
- [X] The code is efficient, to the best of my ability, and does not waste computer resources.
- [X] The code is stable and I have tested it myself, to the best of my abilities.

## Changes:

- Makes the install script use the ZDOTDIR as a destination 

## Other comments:

Closes #9001 
Closes #10479

### CODE REVIEW

**Review by 53jk1 — APPROVED** [review]

**Comment by alexpjohnson** [comment]
@mcornella Would it be possible to get this reviewed?

**Comment by mcornella** [comment]
If `$ZDOTDIR`, we should also update the references to `~/.zshrc`.

File: tools/install.sh [hunk 1]
```diff
@@ -38,7 +38,8 @@
 set -e
 
 # Default settings
-ZSH=${ZSH:-~/.oh-my-zsh}
+ZDOTDIR=${ZDOTDIR:-~}
+ZSH=${ZSH:-${ZDOTDIR}/.oh-my-zsh}
```
Comment (mcornella) [inline tools/install.sh:42]: I'd rather keep this in a single line so we don't define `$ZDOTDIR` and we can later check if it was defined previous to the installer running.

**Review by mcornella — CHANGES_REQUESTED** [review]
The `setup_zshrc` function should be also updated for `$ZDOTDIR` compatibility.

**Pull request closed without merging**